src/cpp/preprocessor/aie2/aie2_blob_preprocessor_input.cpp

```diff
@@ -862,4 +895,100 @@ add_preemption_code(uint32_t col)
     }
     return 0;
   }
+
+
+
+  void
+  config_preprocessor_input::
+  add_pdi(const boost::property_tree::ptree& pdis)
+  {
+    for (auto pd : pdis)
+    {
+      auto pdi = pd.second;
+      std::vector<char> data;
+      readfile(pdi.get<std::string>("PDI_file"), data);
+      m_data[std::string(".pdi.") + pdi.get<std::string>("id")] = data;
+      //std::cout << "pdi name:" << pdi.get<std::string>("id") << std::endl;
+      //std::cout << "pdi file:" << pdi.get<std::string>("PDI_file") << std::endl;
+    }
+  }
+
+  void
+  config_preprocessor_input::
+  add_instance(const boost::property_tree::ptree& pinstance)
+  {
+    for (auto pi : pinstance)
+    {
+      auto pic = pi.second;
+      std::vector<char> data;
+      readfile(pic.get<std::string>("TXN_ctrl_code_file"), data);
+      std::string tname = ".ctrltext." + pic.get<std::string>("id");
+      m_data[tname] = data;
+      //std::cout << "TXN_ctrl_code_file id:" << pic.get<std::string>("id") << std::endl;
+      //std::cout << "TXN_ctrl_code_file:" << pic.get<std::string>("TXN_ctrl_code_file") << std::endl;
+      std::vector<char> cdata;
+      if (!pic.get<std::string>("ctrl_packet_file", "").empty()) {
+        readfile(pic.get<std::string>("ctrl_packet_file"), cdata);
+        m_data[std::string(".ctrldata.") + pic.get<std::string>("id")] = cdata;
+      }
+      std::vector<char> jdata;
+      if (!pic.get<std::string>("patch_info_file", "").empty()) {
+        readfile(pic.get<std::string>("patch_info_file"), jdata);
+        if (jdata.size() !=0 )
```

Comment:
Do we need this special check for zero size? I feel readmetajson should handle zero sized files?